modules/installation-osp-local-disk-deployment.adoc

```diff
@@ -0,0 +1,355 @@
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=jsonpath='{.status.replicas}'=3 controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify the `ClusterOperators` are progressing in the cluster by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait clusteroperators --timeout=30m --all --for=condition=Progressing=false
+----
+
+. Verify that each of the 3 control plane machines has the additional block device you previously created by running the following script:
+
++
+[source,bash]
+----
+$ cp_machines=$(oc get machines -n openshift-machine-api --selector='machine.openshift.io/cluster-api-machine-role=master' --no-headers -o custom-columns=NAME:.metadata.name) <1>
+
+
+if [[ $(echo "${cp_machines}" | wc -l) -ne 3 ]]; then
+  exit 1
+fi <2>
+
+
+for machine in ${cp_machines}; do
+  if ! oc get machine -n openshift-machine-api "${machine}" -o jsonpath='{.spec.providerSpec.value.additionalBlockDevices}' | grep -q 'etcd'; then
+	exit 1
+  fi <3>
+done
+----
+<1> Retrieves the machines running in the cluster.
+<2> Checks that there are 3 machines running.
+<3> Checks that 3 `etcd` devices are running on the 3 machines.
+
+. Create a file named `98-var-lib-etcd.yaml` by using the following content:
+
++
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 98-var-lib-etcd
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - contents: |
+          [Unit]
+          Description=Make File System on /dev/vdb
+          DefaultDependencies=no
+          BindsTo=dev-vdb.device
+          After=dev-vdb.device var.mount
+          Before=systemd-fsck@dev-vdb.service
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+          ExecStart=/usr/sbin/mkfs.xfs -f /dev/vdb
+          TimeoutSec=0
+
+          [Install]
+          WantedBy=var-lib-containers.mount
+        enabled: true
+        name: systemd-mkfs@dev-vdb.service
+      - contents: |
+          [Unit]
+          Description=Mount /dev/vdb to /var/lib/etcd
+          Before=local-fs.target
+          Requires=systemd-mkfs@dev-vdb.service
+          After=systemd-mkfs@dev-vdb.service var.mount
+
+          [Mount]
+          What=/dev/vdb
+          Where=/var/lib/etcd
+          Type=xfs
+          Options=defaults,prjquota
+
+          [Install]
+          WantedBy=local-fs.target
+        enabled: true
+        name: var-lib-etcd.mount
+      - contents: |
+          [Unit]
+          Description=Sync etcd data if new mount is empty
+          DefaultDependencies=no
+          After=var-lib-etcd.mount var.mount
+          Before=crio.service
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+          ExecCondition=/usr/bin/test ! -d /var/lib/etcd/member
+          ExecStart=/usr/sbin/setenforce 0
```

Comment:
We must not do this. Why is this included? @EmilienM?

Reply:
I don't know and I highly don't recommend doing that.
I don't know why it's here, I've asked to look at https://github.com/openshift/installer/blob/master/docs/user/openstack/etcd-ephemeral-disk.md and this doc doesn't have it.

Reply:
The doc you reference does have that @EmilienM https://github.com/openshift/installer/blob/master/docs/user/openstack/etcd-ephemeral-disk.md#:~:text=crio.service%0A%0A%20%20%20%20%20%20%20%20%20%20%5BService%5D-,Type,-%3Doneshot%0A%20%20%20%20%20%20%20%20%20%20RemainAfterExit%3Dyes%0A%20%20%20%20%20%20%20%20%20%20ExecCondition

Reply:
Can someone help me fix the docs, please? I don't write in this area so I'm very unfamiliar with this content. Apologies in advance if I need a bit more support.

Reply:
> The doc you reference does have that
yes, my bad. I was grepping for "selinux" only.

TBH I stole this from https://docs.openshift.com/container-platform/4.14/scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.html#move-etcd-different-disk_recommended-etcd-practices
I'm happy if we improve this, but so far this is what QE and our CI have been testing...

Reply:
We should unpublish all versions of that document 😬  I'll raise a security bug against it.

Reply:
@mdbooth and @EmilienM what needs to change in this document for improvements?